TicTacToe/TicTacToe.cs

```diff
@@ -14,63 +14,102 @@ class Program
 
         public static void Main()
         {
+            int tieInt = 0;     //if reaches 9, game ends
             do
             {
+                int[] newArr = new int[2];
+                DrawBoard();
+                newArr = GetInput();
+                if(PlaceMark(newArr[0],newArr[1]) == true && playerTurn == "X")
+                {
+                    board[newArr[0]][newArr[1]] = playerTurn;
+                    playerTurn = "O";
+                    tieInt++;
+                }
+                else if(PlaceMark(newArr[0],newArr[1]) == true && playerTurn == "O")
+                {
+                    board[newArr[0]][newArr[1]] = playerTurn;
+                    playerTurn = "X";
+                    tieInt++;
+                }
+                else
+                {
+                    newArr = GetInput();        //when user input is already taken or out of bound
+                }
                 DrawBoard();
-                GetInput();
 
-            } while (!CheckForWin() && !CheckForTie());
+            } while (!CheckForWin() && tieInt < 9); //game ending condition
 
             // leave this command at the end so your program does not close automatically
             Console.ReadLine();
         }
-
-        public static void GetInput()
+        
+        public static int[] GetInput()           //return two ints that store user inputs
         {
             Console.WriteLine("Player " + playerTurn);
             Console.WriteLine("Enter Row:");
             int row = int.Parse(Console.ReadLine());
             Console.WriteLine("Enter Column:");
             int column = int.Parse(Console.ReadLine());
+            int[] taken = new[] {row, column};
+            return taken;
         }
-
-        public static void PlaceMark(int row, int column)
+        public static bool PlaceMark(int row, int column) //return true if possible to put a mark
         {
-        // your code goes here
+            // your code goes here
+            if(row > 2 || column > 2)
```

Comment:
since this method is not actually placing the mark, maybe it should be called checkValid, or something similar.